src/a-frame/component-scanner.ts

```diff
@@ -0,0 +1,131 @@
+/**
```

Comment:
could you add some more inline documentation to this file? I haven't used a-frame before and had to make some assumptions about what you were trying to do here

- scanComponent
- enterCallExpression
- other?